src/grid.rs

```diff
@@ -0,0 +1,269 @@
+use std::thread::sleep;
+use std::time::Duration;
+
+pub struct Grid {
+    matrix: Vec<Vec<u8>>,
+    width: usize,
+    height: usize,
+}
+
+impl Grid {
+    pub fn new(width: usize, height: usize) -> Self {
+        let mut cells: Vec<Vec<u8>> = Vec::with_capacity(width);
+        for _ in 0..width {
+            cells.push(vec![0; height]);
+        }
```

Comment:
```suggestion
        let cells: Vec<Vec<u8>> = vec![vec![0; height]; width];
```
I think this looks better.